src/parser.rs

```diff
@@ -932,6 +933,37 @@ impl Parser {
     }
 
     pub fn parse_create_table(&mut self) -> Result<Statement, ParserError> {
+        let if_keyword = self.parse_keyword("IF");
```

Comment:
I think you can simply use `let if_not_exists = self.parse_keywords(vec!["IF", "NOT", "EXISTS"]);`, why not?

Reply:
If I use `let if_not_exists = self.parse_keywords(vec!["IF", "NOT", "EXISTS"]);` I wouldn't be able to say if user typed in just `CREATE TABLE` or missed one of keyword because `self.parse_keywords` returns `bool`

Reply:
Not sure I fully understand what you're saying.. When `parse_keywords` returns `false` the current token is left unchanged. So `CREATE TABLE if (...)` would be accepted (which seems more appropriate for the generic parser at least; I don't know if Postgres rejects that), and missing some of the keywords like `CREATE TABLE IF NOT ...` would fail with an error like "Expected end of statement, found: NOT" instead of more specific error message, which seems acceptable too.

Listing all possible errors for multiple dialects doesn't seem maintainable, and we already use `parse_keywords` in other places.